client/repolist.c

```diff
@@ -438,220 +442,172 @@ TDNFEventRepoReadConfigStart(
-                      &pRepo->pszPass);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        dwError = TDNFReadKeyValueInt(
-                      pSections,
-                      TDNF_REPO_KEY_PRIORITY,
-                      TDNF_REPO_DEFAULT_PRIORITY,
-                      &pRepo->nPriority);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        dwError = TDNFReadKeyValueInt(
-                      pSections,
-                      TDNF_REPO_KEY_TIMEOUT,
-                      TDNF_REPO_DEFAULT_TIMEOUT,
-                      &pRepo->nTimeout);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        dwError = TDNFReadKeyValueInt(
-                      pSections,
-                      TDNF_REPO_KEY_RETRIES,
-                      TDNF_REPO_DEFAULT_RETRIES,
-                      &pRepo->nRetries);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        dwError = TDNFReadKeyValueInt(
-                      pSections,
-                      TDNF_REPO_KEY_MINRATE,
-                      TDNF_REPO_DEFAULT_MINRATE,
-                      &pRepo->nMinrate);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        dwError = TDNFReadKeyValueInt(
-                      pSections,
-                      TDNF_REPO_KEY_THROTTLE,
-                      TDNF_REPO_DEFAULT_THROTTLE,
-                      &pRepo->nThrottle);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        dwError = TDNFReadKeyValueBoolean(
-                      pSections,
-                      TDNF_REPO_KEY_SSL_VERIFY,
-                      TDNF_REPO_DEFAULT_SSLVERIFY,
-                      &pRepo->nSSLVerify);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        dwError = TDNFReadKeyValue(
-                      pSections,
-                      TDNF_REPO_KEY_SSL_CA_CERT,
-                      NULL,
-                      &pRepo->pszSSLCaCert);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        dwError = TDNFReadKeyValue(
-                      pSections,
-                      TDNF_REPO_KEY_SSL_CLI_CERT,
-                      NULL,
-                      &pRepo->pszSSLClientCert);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        dwError = TDNFReadKeyValue(
-                      pSections,
-                      TDNF_REPO_KEY_SSL_CLI_KEY,
-                      NULL,
-                      &pRepo->pszSSLClientKey);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        dwError = TDNFReadKeyValue(
-                      pSections,
-                      TDNF_REPO_KEY_METADATA_EXPIRE,
-                      TDNF_REPO_DEFAULT_METADATA_EXPIRE_STR,
-                      &pszMetadataExpire);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        dwError = TDNFParseMetadataExpire(
-                      pszMetadataExpire,
-                      &pRepo->lMetadataExpire);
-        BAIL_ON_TDNF_ERROR(dwError);
-
-        TDNF_SAFE_FREE_MEMORY(pszMetadataExpire);
-        pszMetadataExpire = NULL;
+        if (errno != 0)
+        {
+            dwError = errno;
+            BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+        }
+        else
```

Comment:
condition is not required

Reply:
Can you elaborate? Which condition is not needed and why?

Reply:
Sorry, my bad BAIL_ON_TDNF_SYSTEM_ERROR and BAIL_ON_TDNF_ERROR